### ISSUE TITLE
Add tf.nn.log_gaussian_loss

### ISSUE THREAD

**Issue by nailimixaM** [tensorflow/tensorflow#47186]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Implement the Gaussian negative log-likelihood loss for the tf.nn class, similar to the [log poisson loss](https://www.tensorflow.org/api_docs/python/tf/nn/log_poisson_loss). See also the [PyTorch version](https://pytorch.org/docs/master/generated/torch.nn.GaussianNLLLoss.html). 

**Will this change the current api? How?**
tf.nn class will have the log_gaussian_loss function, which could look like:
`tf.nn.log_gaussian_loss(
    targets, inputs, variances, compute_full_loss=False, name=None
)`
to keep in line with the `log_poisson_loss` implementation.

**Who will benefit with this feature?**
This loss function is used extensively in Bayesian deep ensembles, see for example [Lakshminarayanan et al. 2017](https://scholar.google.co.uk/citations?user=QYn8RbgAAAAJ&hl=en&oi=sra#d=gs_md_cita-d&u=%2Fcitations%3Fview_op%3Dview_citation%26hl%3Den%26user%3DQYn8RbgAAAAJ%26citation_for_view%3DQYn8RbgAAAAJ%3AhCrLmN-GePgC%26tzom%3D0), though the loss function has been around since the 90s (Mean and Variance Estimation, [Nix et al. 1994](https://ieeexplore.ieee.org/document/374138)).

**Any Other info.**

**Comment by nailimixaM** [comment]
Hi @rmothukuru, any update on this? Thanks

**Comment by around-star** [comment]
@nailimixaM , @rmothukuru if I am not wrong, isn't the one you are looking for is [sigmoid_cross_entropy_with_logits](https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/ops/nn_impl.py#L115-L191)

**Comment by nailimixaM** [comment]
Hi @around-star, I'm afraid that's not it, this loss function is for regression, not classification

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
